lib/pbench/server/api/resources/upload_api.py

```diff
@@ -59,6 +60,9 @@ def __init__(self, config, logger):
 
     @Auth.token_auth.login_required()
     def put(self, filename: str):
+        disabled = ServerConfig.get_disabled()
+        if disabled:
+            abort(HTTPStatus.SERVICE_UNAVAILABLE, **disabled)
```

Comment:
I'm really wishing that we had a "This is a Pbench Server API function" `@`decorator, so that we didn't have to add this code to every API function (i.e., this time we would just have to add the decorator...and, next time we wanted to do something for all the API functions, we could do it in one place...).

Reply:
There are a few APIs, including the user endpoints (which will go away) and `PUT`, which were written before `ApiBase` and not retrofitted. As `ApiBase._dispatch` will take care of this automatically, my preferred solution would be to refactor the `upload_api.py` on top of `ApiBase`. I just didn't want to throw that into this PR, so the special case was more or less unavoidable.